### DISCUSSION TITLE
Installing in linux

### DISCUSSION THREAD

**Discussion by mwwiley** [pangeo-data/xESMF#239] [Q&A]
Hello, I have been having a very hard time getting xESMF to work on a linux instance.  I was able to use/install in a jupyter notebook on MacOS without too much trouble, but I am trying to migrate my process to an AWS EC2 instance.  I am running Ubuntu 16  (I know it's old but there are other reasons).   I am in a conda environment using python 3.10.9.  I was able to download and compile and install ESMF from source (version 8.4.0).  I was able to 'conda install esmpy'

>import esmpy
>!echo $ESMFMKFILE
`/home/efs/system/jupyter/esmf-8.4.0/DEFAULTINSTALLDIR/lib/libO/Linux.gfortran.64.mpiuni.default/esmf.mk`

'conda install xesmf' appears to run fine, but then....
> import xesmf
```
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[6], line 1
----> 1 import xesmf
File /home/efs/system/anaconda3/envs/primo/lib/python3.10/site-packages/xesmf/__init__.py:4
      2 from . import util
      3 from . import data
----> 4 from . frontend import Regridder

File /home/efs/system/anaconda3/envs/primo/lib/python3.10/site-packages/xesmf/frontend.py:10
      7 import os
      8 import warnings
---> 10 from . backend import (esmf_grid, esmf_locstream, add_corner,
     11                        esmf_regrid_build, esmf_regrid_finalize)
     13 from . smm import read_weights, apply_weights
     15 try:

File /home/efs/system/anaconda3/envs/primo/lib/python3.10/site-packages/xesmf/backend.py:19
      1 '''
      2 Backend for xESMF. This module wraps ESMPy's complicated API and can create
      3 ESMF Grid and Regrid objects only using basic numpy arrays.
   (...)
     15 So it would be helpful to catch some common mistakes in Python level.
     16 '''
     18 import numpy as np
---> 19 import ESMF
     20 import warnings
     21 import os

ModuleNotFoundError: No module named 'ESMF'
```

Question:   What is the ESMF module if not esmpy or ESMF?   Is this just a PATH issue?   Is ESMF an alias?   I am regridding lambert conformal conic data from an NWP model to match a different resolution lcc grid and have found xesmf to be very fast and efficient.  I would love to get this working and not have to reinvent the wheel!

Any help appreciated!

**Comment by mwwiley** [comment]
Found this in ESMF release notes
`ESMPy users please note that the Python module has been renamed from "ESMF" to "esmpy" for better alignment with [Python Enhancement Proposal (PEP)](https://peps.python.org/) guidelines. Please check out the respective release note for details and potential impact on user code.`
So is this a version issue?

**Comment by mwwiley** [comment]
changing line 19 of xesmf/backend.py from 
`import ESMF`
to 
`import esmpy` 
seems to have fixed it.  Leaving this thread here in case it helps others.

**Comment by raphaeldussin** [comment]
@mwwiley are you using the latest release? this issue has been fixed by #214

**Reply by mwwiley** [reply]
I am using 0.7.0

**Reply by raphaeldussin** [reply]
```
(woa18remap) raphael@xps:~/dev/remap_WOA18$ conda list | grep esm
esmf                      8.4.0           mpi_mpich_hc592774_104    conda-forge
esmpy                     8.4.0           mpi_mpich_py39h3088dd8_102    conda-forge
xesmf                     0.7.0              pyhd8ed1ab_0    conda-forge
(woa18remap) raphael@xps:~/dev/remap_WOA18$ ipython
Python 3.9.16 | packaged by conda-forge | (main, Feb  1 2023, 21:39:03) 
Type 'copyright', 'credits' or 'license' for more information
IPython 8.10.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import xesmf

In [2]: 
```

sources in version 0.7.0 have that fix, did you install xesmf with `conda install -c conda-forge xesmf` ?

**Reply by mwwiley** [reply]
That is the exact command I used, I have been battling conda constantly trashing my environment, so I may well have messed something up.  Conda tried to push me to python 3.11.0 which didn't make anything happy.
